src/topology/homotopy/induced_maps.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+import topology.homotopy.product
+import topology.homotopy.equiv
+import topology.homotopy.casts
+import category_theory.equivalence
+
+/-!
+# Homotopic maps induce naturally isomorphic functors
+
+## Main definitions
+
+  - `fundamental_groupoid_functor.homotopic_maps_nat_iso H` The natural isomorphism
+    between the induced functors `f : π(X) ⥤ π(Y)` and `g : π(X) ⥤ π(Y)`, given a homotopy
+    `H : f ∼ g`
+
+  - `fundamental_groupoid_functor.equiv_of_homotopy_equiv hequiv` The equivalence of the categories
+    `π(X)` and `π(Y)` given a homotopy equivalence `hequiv : X ≃ₕ Y` between them.
+
+## Implementation notes
+  - In order to be more universe polymorphic, we define `continuous_map.homotopy.ulift_map`
+  which lifts a homotopy from `I × X → Y` to `(Top.of ((ulift I) × X)) → Y`
+-/
+
+noncomputable theory
+
+universe u
+
+namespace unit_interval
+open_locale unit_interval
+
+/-- The path 0 ⟶ 1 in I -/
+def path01 : path (0 : I) 1 := { to_fun := id, source' := rfl, target' := rfl }
+
+/-- The path 0 ⟶ 1 in ulift I -/
+def upath01 : path (ulift.up 0 : ulift.{u} I) (ulift.up 1) :=
+{ to_fun := ulift.up, source' := rfl, target' := rfl }
+
+local attribute [instance] path.homotopic.setoid
+/-- The homotopy path class of 0 → 1 in `ulift I` -/
+def uhpath01 := @fundamental_groupoid.from_path (Top.of $ ulift.{u} I) _ _ ⟦upath01⟧
+
+end unit_interval
+
+namespace continuous_map.homotopy
+open fundamental_groupoid
+open category_theory
+open unit_interval (uhpath01)
+open fundamental_groupoid_functor
+
+open_locale fundamental_groupoid
+open_locale unit_interval
```

Comment:
You should open those all at once at the top of the file (I haven't checked but I assume it's unlikely you get a conflict by opening slightly more than you need within each section).